LEGO1/mxobjectfactory.h

```diff
@@ -2,11 +2,33 @@
 #define MXOBJECTFACTORY_H
 
 #include "mxcore.h"
+#include "mxatomid.h"
+
+#define FOR_MXOBJECTFACTORY_OBJECTS(X)  \
+  X(MxPresenter)                        \
+  X(MxCompositePresenter)               \
+  X(MxVideoPresenter)                   \
+  X(MxFlcPresenter)                     \
+  X(MxSmkPresenter)                     \
+  X(MxStillPresenter)                   \
+  X(MxWavePresenter)                    \
+  X(MxMIDIPresenter)                    \
+  X(MxEventPresenter)                   \
+  X(MxLoopingFlcPresenter)              \
+  X(MxLoopingSmkPresenter)              \
+  X(MxLoopingMIDIPresenter)
 
 // VTABLE 0x100dc220
 class MxObjectFactory : public MxCore
 {
-
+public:
+  MxObjectFactor();
+  virtual MxCore *Create(const char *name); // vtable 0x14
+  virtual void vtable18(void *); // vtable 0x18
+private:
+#define X(V) MxAtomId m_id##V;
+  FOR_MXOBJECTFACTORY_OBJECTS(X)
```

Comment:
Tip: After the type you might want to add compile-time assert for `offsetof` one or more of these fields. Better safe than sorry.